tsdb/engine/tsm1/wal.go

```diff
@@ -27,6 +27,9 @@ const (
 
 	defaultBufLen = 1024 << 10 // 1MB (sized for batches of 5000 points)
 
+	// walEncodeBufSize is the size of the wal entry encoding buffer
+	walEncodeBufSize = 4 * 1024 * 1024
```

Comment:
Previously the code was allocating a new 1MB slice each time.  With large numbers and sizes of keys, we'd end up growing it all the time which created a ton of garbage.  I'll probably make this a config var in another PR since someone with 10M keys or really long keys may exceed this buffer size still.